Model/Api/OAuthRedirect.php

```diff
@@ -221,9 +241,22 @@ public function login($code = '', $scope = '', $state = '')
         }
 
         try {
+            $shouldLinkOrder = $order_id !== '' && $customer === null;
+
             $customer = $customer ?: $this->createNewCustomer($websiteId, $storeId, $payload);
+
+            if ($shouldLinkOrder) {
+                $order = $this->cartHelper->getOrderByIncrementId($order_id);
+                if ($order !== false && !$order->getCustomerId() && $order->getBillingAddress()->getEmail() === $payload['email']) {
+                    $order->setCustomerId($customer->getId());
+                    $order->setCustomerIsGuest(0);
+                    $this->orderRepository->save($order);
+                }
+            }
+
             $this->linkLoginAndRedirect($externalID, $customer->getId());
         } catch (Exception $e) {
+            $this->bugsnag->notifyError('OAuthRedirect', $e->getMessage());
```

Comment:
It's better to send all exception object into bugsnag, not only message. In this case, we will know the exact line where error happened.

Reply:
will update

Reply:
updated.